README.md

```diff
@@ -0,0 +1,843 @@
+    <a11y-input .label=${'foo'}></a11y-input>
+  `));
+  expect(el).lightDom.to.equal(`
+    <label slot="label">foo</label>
+    <input slot="input">
+  `);
+});
+```
+
+And let's implement it.
+
+```js
+connectedCallback() {
+  super.connectedCallback();
+  this.labelEl = document.createElement('label');
+  this.labelEl.innerText = this.label;
+  this.labelEl.setAttribute('slot', 'label');
+  this.appendChild(this.labelEl);
+
+  this.inputEl = document.createElement('input');
+  this.inputEl.setAttribute('slot', 'input');
+  this.appendChild(this.inputEl);
+}
+```
+
+*yes that is sort of an anti-pattern however to allow for a11y it might be a real use case - anyways it's great for illustration purposes*
+
+Now, this leads to a certain problem - can you guess it?
+
+### My Filter App
+
+So now that we have a basic a11y input let's use it in our application.
+
+Again we start with a skeleton `src/my-app.js`
+```js
+/* eslint-disable class-methods-use-this */
+import { LitElement, html, css } from 'lit-element';
+
+export class MyApp extends LitElement {}
+
+customElements.define('my-app', MyApp);
+```
+
+tests `test/my-app.test.js`;
+```js
+/* eslint-disable no-unused-expressions */
+import { html, fixture, expect } from '@open-wc/testing';
+
+import '../src/my-app.js';
+
+/**
+ * @typedef {import('../src/my-app.js').MyApp} MyApp
+ */
+
+describe('My Filter App', () => {
+  it('has a heading and a search field', async () => {
+    const el = /** @type {MyApp} */ (await fixture(html`
+      <my-app .label=${'foo'}></my-app>
+    `));
+    expect(el).shadowDom.to.equal(`
+      <h1>My Filter App</h1>
+      <a11y-input></a11y-input>
+    `);
+  });
+});
+```
+
+and then we fill
+```js
+render() {
+  return html`
+    <h1>My Filter App</h1>
+    <a11y-input></a11y-input>
+  `;
+}
+```
+
+and oh no... that should be green...
+
+```
+SUMMARY:
+✔ 3 tests completed
+✖ 1 test failed
+
+FAILED TESTS:
+  My Filter App
+    ✖ has a heading and a search field
+    AssertionError: expected '<h1>\n  My Filter App\n</h1>\n<a11y-input>\n  <label slot="label">\n  </label>\n  <input slot="input">\n</a11y-input>\n' to equal '<h1>\n  My Filter App\n</h1>\n<a11y-input>\n</a11y-input>\n'
+
+      + expected - actual
+
+       <h1>
+         My Filter App
+       </h1>
+       <a11y-input>
+      -  <label slot="label">
+      -  </label>
+      -  <input slot="input">
+       </a11y-input>
+```
+
+So I'm only interested in `a11y-input` not in its implementation detail.
+As an app developer, I'm now a little unlucky as this element "plays" in my domain. preposterous!
+After some investigation I conclude there are very valid reasons for a11y input to do what it does.
+Now how can I test it?
+
+Luckily `.shadowDom` has another ace up its sleeve.
+As it allows us to ignore certain dom parts.
+
+```js
+expect(el).shadowDom.to.equal(
+  `
+  <h1>My Filter App</h1>
+  <a11y-input></a11y-input>
+`,
+  { ignoreChildren: ['a11y-input'] },
```

Comment:
I think we could move this to an advanced section. If you are new to web compoments you dont need this

Reply:
this whole article is an advanced section 🙈 

added a section at the top asking if there is a need for a simpler version of this post then they should just let us know in the comments 🤗